internal/driver/basenotificationresthandler.go

```diff
@@ -0,0 +1,145 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+
+	"github.com/IOTechSystems/onvif/event"
+	"github.com/IOTechSystems/onvif/gosoap"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	OnvifEventRestPath = "onvifevent"
+	apiResourceRoute   = common.ApiBase + "/" + OnvifEventRestPath + "/{" + common.DeviceName + "}/{" + common.ResourceName + "}"
+	handlerContextKey  = "RestHandler"
+)
+
+type RestHandler struct {
+	service     *sdk.DeviceService
+	logger      logger.LoggingClient
+	asyncValues chan<- *models.AsyncValues
+}
+
+func NewRestHandler(service *sdk.DeviceService, logger logger.LoggingClient, asyncValues chan<- *models.AsyncValues) *RestHandler {
+	handler := RestHandler{
+		service:     service,
+		logger:      logger,
+		asyncValues: asyncValues,
+	}
+
+	return &handler
+}
+
+func (handler RestHandler) Start() errors.EdgeX {
+	if err := handler.service.AddRoute(apiResourceRoute, handler.addContext(deviceHandler), http.MethodPost); err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("unable to add required route: %s: %s", apiResourceRoute, err.Error()), err)
+	}
+
+	handler.logger.Infof("Route %s added.", apiResourceRoute)
+	return nil
+}
+
+func (handler RestHandler) addContext(next func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
+	// Add the context with the handler so the endpoint handling code can get back to this handler
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := context.WithValue(r.Context(), handlerContextKey, handler)
+		next(w, r.WithContext(ctx))
+	})
+}
```

Comment:
This is overly complex. I can't determine the actual handler that is being used.
```suggestion
	if err := handler.service.AddRoute(apiResourceRoute, handler.processAsyncRequest, http.MethodPost); err != nil {
		return errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("unable to add required route: %s: %s", apiResourceRoute, err.Error()), err)
	}

	handler.logger.Infof("Route %s added.", apiResourceRoute)
	return nil
}
```